### PR TITLE
[#1235] ported statemachine example to use java instead of xtend

### PR DESCRIPTION
**Pull request by cdietrich**
[#1235] ported statemachine example to use java instead of xtend
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/folding/StatemachineFoldingRegionProvider.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		}
+	}
+
+	/**
+	 * Determine the text region between the start keyword and the end keyword.
+	 */
+	private ITextRegion textRegionBetween(XtextResource res, Keyword startKeyword, Keyword endKeyword) {
+		INode startNode = null;
+		for (INode node : res.getParseResult().getRootNode().getAsTreeIterable()) {
+			{
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/folding/StatemachineFoldingRegionProvider.java]: Unnecessary block

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/folding/StatemachineFoldingRegionProvider.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+				EObject grammarElement = node.getGrammarElement();
+				if (Objects.equal(grammarElement, startKeyword)) {
+					startNode = node;
+				}
+				if (Objects.equal(grammarElement, endKeyword) && startNode != null) {
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/folding/StatemachineFoldingRegionProvider.java]: Could use identity equals `==`

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineLabelProvider.java [hunk 3]
```diff
@@ -15,17 +15,17 @@
...
-    return ("state " + _name);
-  }
+
+	@Inject
+	public StatemachineLabelProvider(AdapterFactoryLabelProvider delegate) {
+		super(delegate);
+	}
+
+	public String text(State state) {
+		return ("state " + state.getName());
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineLabelProvider.java]: Unnecessary parenthesis

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/quickfix/StatemachineQuickfixProvider.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ */
+public class StatemachineQuickfixProvider extends DefaultQuickfixProvider {
+	@Fix(StatemachineValidator.INVALID_NAME)
+	public void convertNameToFirstLowerCase(final Issue issue, final IssueResolutionAcceptor acceptor) {
+		String[] data = issue.getData();
+		String firstLower = StringExtensions.toFirstLower(data[0]);
+		String label = "Change to \'" + firstLower + "\'.";
+		acceptor.accept(issue, label, label, "upcase.png", (IModification) (IModificationContext ctx) -> {
+			IXtextDocument xtextDocument = ctx.getXtextDocument();
+			String firstLetter = xtextDocument.get(issue.getOffset().intValue(), 1);
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/quickfix/StatemachineQuickfixProvider.java]: Unnecessary explicit unboxing

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/folding/StatemachineFoldingRegionProvider.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+			computeResetEventsFolding(res, foldingRegionAcceptor);
+			computeCommandsFolding(res, foldingRegionAcceptor);
+		} else {
+			super.computeObjectFolding(o, foldingRegionAcceptor);
+		}
+	}
+
+	@Override
+	protected boolean isHandled(EObject o) {
+		return (super.isHandled(o) || o instanceof Statemachine);
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/folding/StatemachineFoldingRegionProvider.java]: No need for brackets

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineDescriptionLabelProvider.java [hunk 6]
```diff
@@ -1,30 +1,28 @@
...
  */
-class StatemachineDescriptionLabelProvider extends DefaultDescriptionLabelProvider {
+public class StatemachineDescriptionLabelProvider extends DefaultDescriptionLabelProvider {
 
 	// Labels and icons can be computed like this:
-
-//	override text(IEObjectDescription ele) {
-//		ele.name.toString
+	
+//	String text(IEObjectDescription ele) {
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineDescriptionLabelProvider.java]: Access modifier? `@Override`?

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineDescriptionLabelProvider.java [hunk 7]
```diff
@@ -1,30 +1,28 @@
...
+//		return ele.getName().toString();
 //	}
-//
-//	override image(IEObjectDescription ele) {
-//		ele.EClass.name + '.gif'
+//	 
+//	String image(IEObjectDescription ele) {
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineDescriptionLabelProvider.java]: Access modifier? `@Override`?

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/quickfix/StatemachineQuickfixProvider.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Custom quickfixes.
+ *
+ * See
+ * https://www.eclipse.org/Xtext/documentation/310_eclipse_support.html#quick-fixes
+ */
+public class StatemachineQuickfixProvider extends DefaultQuickfixProvider {
+	@Fix(StatemachineValidator.INVALID_NAME)
+	public void convertNameToFirstLowerCase(final Issue issue, final IssueResolutionAcceptor acceptor) {
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/quickfix/StatemachineQuickfixProvider.java]: final

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl/src/org/eclipse/xtext/example/fowlerdsl/GenerateStatemachine.mwe2 [hunk 9]
```diff
@@ -32,6 +32,7 @@ Workflow {
 				createEclipseMetaData = true
 			}
 			code = {
+				preferXtendStubs = false
 				encoding = "UTF-8"
 				lineDelimiter = "\n"
 				fileHeader = "/*\n * generated by Xtext\n */"
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl/src/org/eclipse/xtext/example/fowlerdsl/GenerateStatemachine.mwe2]: No need for the fileheader to change here?

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl/src/org/eclipse/xtext/example/fowlerdsl/GenerateStatemachine.mwe2]: this was inconsistent an i want to do this in a separate pr

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineDescriptionLabelProvider.java (hunk 7 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineDescriptionLabelProvider.java]: this is the default template. please create an issue against the generator

File: org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineDescriptionLabelProvider.java (hunk 6 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/projects/fowlerdsl/org.eclipse.xtext.example.fowlerdsl.ui/src/org/eclipse/xtext/example/fowlerdsl/ui/labeling/StatemachineDescriptionLabelProvider.java]: this is the default template. please create an issue against the generator

**Pull request merged**